StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1404CodeAnalysisSuppressionMustHaveJustification.cs

```diff
@@ -49,7 +53,45 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxNodeAction(HandleAttributeNode, SyntaxKind.Attribute);
+        }
+
+        private void HandleAttributeNode(SyntaxNodeAnalysisContext context)
+        {
+            var attribute = context.Node as AttributeSyntax;
+            if(attribute != null)
+            {
+                SymbolInfo symbolInfo = context.SemanticModel.GetSymbolInfo(attribute);
+                ISymbol symbol = symbolInfo.Symbol;
+                if(symbol != null)
+                {
+                    var suppressMessageType = context.SemanticModel.Compilation.GetTypeByMetadataName(typeof(SuppressMessageAttribute).FullName);
+                    if (symbol.ContainingType == suppressMessageType)
+                    {
+
+                        foreach (var argument in attribute.ArgumentList.ChildNodes())
+                        {
+                            var attributeArgument = argument as AttributeArgumentSyntax;
+                            if (attributeArgument?.NameEquals?.Name?.ToString() == "Justification")
+                            {
+                                // Check if the justification is not empty
+
+                                var value = context.SemanticModel.GetConstantValue(attributeArgument.Expression);
+
+                                // If value does not have a value the expression is not constant -> Compilation error
```

Comment:
What happens if the user passes a constant which is not a string? The code won't compile, but would it cause an exception for the cast to `string`?